### PR TITLE
[RW-3856][risk=no] Fix workspace list filtering on reload

### PR DESCRIPTION
**Pull request by calbach**
Add test coverage of workspace list filtering

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-list.spec.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
-import {mount} from 'enzyme';
+import { mount, ReactWrapper} from 'enzyme';
```
Comment (dolbeew) [inline ui/src/app/pages/workspace/workspace-list.spec.tsx:1]: Nit: Seems likely this whitespace added after the opening brace was unintended.

File: ui/src/app/pages/workspace/workspace-list.spec.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 import * as React from 'react';
 
 import {
   registerApiClient
 } from 'app/services/swagger-fetch-clients';
-import {Profile, ProfileApi, WorkspacesApi} from 'generated/fetch';
+import { Profile, ProfileApi, WorkspaceAccessLevel, WorkspacesApi} from 'generated/fetch';
```
Comment (dolbeew) [inline ui/src/app/pages/workspace/workspace-list.spec.tsx:7]: Nit: Remove whitespace if unintended.

File: ui/src/app/pages/workspace/workspace-list.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import {WorkspacePermissions} from 'app/utils/workspace-permissions';
 import * as React from 'react';
 import RSelect from 'react-select';
 import * as fp from 'lodash/fp';
-import {Profile} from 'generated/fetch';
+import { Profile, WorkspaceAccessLevel} from 'generated/fetch';
```
Comment (dolbeew) [inline ui/src/app/pages/workspace/workspace-list.tsx:19]: Nit: Remove whitespace if unintended.

**Review by dolbeew — APPROVED** [review]

File: ui/src/app/pages/workspace/workspace-list.spec.tsx (hunk 1 above)
Comment (calbach) [reply ui/src/app/pages/workspace/workspace-list.spec.tsx]: fixed - this is definitely something I want to cover with the linter soon

**Review by calbach — COMMENTED** [review]
thanks

**Pull request merged**